File: Unidad 4/C#/P17_U4.cs
using System;
class P17_U4 {
  static void Main() {
            Console.Write("Ingrese el número de filas de las matrices: ");
            int filas = Convert.ToInt32(Console.ReadLine());

            Console.Write("Ingrese el número de columnas de las matrices: ");
            int columnas = Convert.ToInt32(Console.ReadLine());

            int[,] matrizA = new int[filas, columnas];
            int[,] matrizB = new int[filas, columnas];
            int[,] resultado = new int[filas, columnas];

            Console.WriteLine("Ingrese los valores para la matriz A:");
            Sol_Valores(matrizA);

            Console.WriteLine("Ingrese los valores para la matriz B:");
            Sol_Valores(matrizB);

            Mult_Matrices(matrizA, matrizB, resultado);

            Console.WriteLine("La matriz resultante es:");
            Mostrar_Matriz(resultado);
        }

        private static void Sol_Valores(int[,] matriz)
        {
            for (int i = 0; i < matriz.GetLength(0); i++)
            {
                for (int j = 0; j < matriz.GetLength(1); j++)
                {
                    Console.Write($"Ingrese el valor en la posición [{i}][{j}]: ");
                    matriz[i, j] = Convert.ToInt32(Console.ReadLine());
                }
            }
        }

        private static void Mult_Matrices(int[,] matrizA, int[,] matrizB, int[,] resultado)
        {
            for (int i = 0; i < matrizA.GetLength(0); i++)
            {
                for (int j = 0; j < matrizB.GetLength(1); j++)
                {
                    for (int k = 0; k < matrizA.GetLength(1); k++)
                    {
                        resultado[i, j] += matrizA[i, k] * matrizB[k, j];
                    }
                }
            }
        }

        private static void Mostrar_Matriz(int[,] matriz)
        {
            for (int i = 0; i < matriz.GetLength(0); i++)
            {
                for (int j = 0; j < matriz.GetLength(1); j++)
                {
                    Console.Write(matriz[i, j] + " ");
                }
                Console.WriteLine();
            }
        }
    }

File: Unidad 4/JAVA/P17_U4.java
public class P17_U4
{
	public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese el número de las filas de las matrices: ");
        int filas = scanner.nextInt();

        System.out.print("Ingrese el número de las columnas de las matrices: ");
        int columnas = scanner.nextInt();

        int[][] matrizA = new int[filas][columnas];
        int[][] matrizB = new int[filas][columnas];
        int[][] resultado = new int[filas][columnas];

        System.out.println("Ingrese los valores para la matriz A:");
        sol_Valores(matrizA, scanner);

        System.out.println("Ingrese los valores para la matriz B:");
        sol_Valores(matrizB, scanner);

        mult_Matrices(matrizA, matrizB, resultado);

        System.out.println("La matriz resultante es:");
        mostrar_Matriz(resultado);

        scanner.close();
    }

    private static void sol_Valores(int[][] matriz, Scanner scanner) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print("Ingrese el valor en la posición [" + i + "][" + j + "]: ");
                matriz[i][j] = scanner.nextInt();
            }
        }
    }

    private static void mult_Matrices(int[][] matrizA, int[][] matrizB, int[][] resultado) {
        for (int i = 0; i < matrizA.length; i++) {
            for (int j = 0; j < matrizB[0].length; j++) {
                for (int k = 0; k < matrizA[0].length; k++) {
                    resultado[i][j] += matrizA[i][k] * matrizB[k][j];
                }
            }
        }
    }

    private static void mostrar_Matriz(int[][] matriz) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print(matriz[i][j] + " ");
            }
            System.out.println();
        }
    }
}